napari/_qt/qt_viewer.py

```diff
@@ -285,6 +284,61 @@ def dropEvent(self, event):
                 filenames.append(path)
         self._add_files(filenames)
 
+    def _add_files(self, filenames):
+        """Adds an image/volume layer to the viewer.
+
+        Whether the image is multichannel is determined by
+        :func:`napari.util.misc.is_multichannel`.
+
+        If multiple images are selected, they are stacked along the 0th
+        axis.
+
+        If there is only 1 filename opened and it is a 3D numpy array saved
+        in either npy or npz format, a Volume layer is added and the camera is
+        changed to a 3D camera called TurntableCamera.
+
+        TODO Add event to switch between 3 kinds of cameras that vispy offers
+
+        Parameters
+        -------
+        filenames : list
+            List of filenames to be opened
+        """
+
+        if len(filenames) == 1 and (
+            filenames[0].endswith(".npy") or filenames[0].endswith(".npz")
+        ):
+            volume = load_numpy_array(filenames[0])
+
+            # Create an XYZaxis visual
+            self.axis = XYZAxis(parent=self.view)
+            self.axis.transform = STTransform(
+                translate=(50, 50), scale=(50, 50, 50, 1)
+            ).as_matrix()
+
+            # Set a 3D camera
+            self.view.camera = TurntableCamera(name="TurntableCamera")
+            # flip y-axis to have correct alignment
+            self.view.camera.flip = (0, 1, 0)
+            self.view.camera.set_range()
+            self.view.camera.viewbox_key_event = viewbox_key_event
+
+            # TO DO: Remove
+            self.viewer._view = self.view
+
+            self.viewer.add_volume(
+                volume, multichannel=is_multichannel(volume.shape)
```

Comment:
multichannel will always be `FALSE` for volumetric data as vispy doesn't support rendering RGB or RGBA data.